### ISSUE TITLE
pkgcraft-tools: improve '--help

### ISSUE THREAD

**Issue by idealseal** [pkgcraft/pkgcraft#217]

**Comment by radhermit** [comment]
Yes, unfortunately documentation hasn't been a high priority so far especially for pkgcraft-tools. I do agree that the descriptions for many of the commands are currently lacking along with needing some form of manually written docs (https://github.com/pkgcraft/pkgcraft/issues/104).

For a brief description here, pkgcraft-tools are somewhat meant to be akin to what portage-utils provides in a more portage-centric manner in that they are CLI tools mainly targeting scripting and development. For example, `pk dep ...` provides somewhat similar functionality to `qatom`.

Additionally, one command that currently might be the most useful (depending on how you consume ebuild repos) would be `pk repo metadata regen` which performs the same ebuild metadata regeneration action as `egencache` or `emerge --regen` from portage and `pmaint regen` from pkgcore but at a much higher performance. See the [blog post ](https://pkgcraft.github.io/posts/metadata-cache-generation/) about pkgcraft's metadata cache generation support for more information and benchmarks if you're interested.

**Comment by idealseal** [comment]
I'm not opposed to contributing some of the necessary changes to make clap output nicer. The most important thing for me is discoverability though the command line as it's the most accessible thing to get to know what each command does. I was mostly struggling to figure out what `cpv` is or does.

> For a brief description here, pkgcraft-tools are somewhat meant to be akin to what portage-utils provides in a more portage-centric manner in that they are CLI tools mainly targeting scripting and development. For example, pk dep ... provides somewhat similar functionality to qatom.

A while ago I had similar ideas, but I was writing code before planning properly. I started with parsing the local package database (so a `qlist -I` / `eix`  replacement), but I was frustrated with the lack of performance because of the atom regex parsing. I just think the design over all wasn't all that good.

Another thing:  Is it correct that every command is split up into sub modules? I find it really hard to navigate (on GitHub at least) and I really like the `cli` or whatever module to be in a single file to have something like a command reference.

**Comment by radhermit** [comment]
> I'm not opposed to contributing some of the necessary changes to make clap output nicer. The most important thing for me is discoverability though the command line as it's the most accessible thing to get to know what each command does. I was mostly struggling to figure out what `cpv` is or does.

A `cpv` is simplified reference for a singular package in the `category/package-version` format (e.g. cat/pkg-1-r2) as opposed to `pk dep` which operates on the full (and extended) dependency specification (depspec) format supporting operators, use flags, repo specifier, etc (e.g. `>=cat/pkg-1-r2::repo[u1?,u2(+)]`).

> A while ago I had similar ideas, but I was writing code before planning properly. I started with parsing the local package database (so a `qlist -I` / `eix` replacement), but I was frustrated with the lack of performance because of the atom regex parsing. I just think the design over all wasn't all that good.

Yes, using regex's will not be very performant overall which may or may not matter depending on the target use cases. Also, properly implementing the depspec format using regular expressions requires using positive/negative lookahead in some form which hurts performance even more.

> Another thing: Is it correct that every command is split up into sub modules? I find it really hard to navigate (on GitHub at least) and I really like the `cli` or whatever module to be in a single file to have something like a command reference.

It's mostly just a code layout choice I've currently made since some subcommands (especially future ones) may get very complex and having multiple of those in a single file will get messy in my opinion.
